### ISSUE TITLE
Try different tricks for DNN with MNIST

### ISSUE THREAD

**Issue by anton-dergunov** [anton-dergunov/ml-playground#12]
Implement DNN with 5 hidden layers (100 neurons each) and try different tricks:
* Xavier, He and just 0 weight initialization
* Nonsaturating activation functions
  * Leaky ReLU (and randomized leaky, parametric leaky)
  * ELU
  * tanh
  * logistic
* Batch normalization
* Different optimizers
  * Momentum optimization
  * Nesterov Accelerated Gradient
  * AdaGrad
  * RMSProp
  * Adam
  * Compare speed and performance on train & test
* l1 and l2 regularization
* Dropout (and compare overfitting)
* Max-Norm Regularization
* Data Augmentation
